libraries/TGRSIAnalysis/TS3/TS3.cxx

```diff
@@ -128,14 +128,16 @@ void TS3::BuildPixels(){
 						EneS[j]*fFrontBackEnergy<EneR[i]){  //if time is good check energy
 
 						//Now we have accepted a good event, build it
-						TS3Hit dethit = fS3RingHits[i]; // Ring defines all data sector just gives position
-						dethit.SetSectorNumber(fS3SectorHits[j].GetSector());
-// 						if(TGRSIRunInfo::IsWaveformFitting()){
-// 							dethit.SetTimeFit(fS3RingHits[i].GetFitTime());
-// 							dethit.SetSig2Noise(fS3RingHits[i].GetSignalToNoise());
-// 						}
-						fS3Hits.push_back(dethit);
-
+						if(SectorPreference()){
+							TS3Hit dethit = fS3SectorHits[j]; // Sector defines all data ring just gives position
+							dethit.SetRingNumber(fS3RingHits[i].GetRing());
+							fS3Hits.push_back(dethit);
```

Comment:
I'd leave the push_back on the outside of the if statement since it is called both times anyway